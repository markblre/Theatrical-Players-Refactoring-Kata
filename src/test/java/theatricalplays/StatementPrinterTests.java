package theatricalplays;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.approvaltests.Approvals.verify;
import static org.approvaltests.Approvals.verifyHtml;

public class StatementPrinterTests {

  @Test
  void testTXTInvoiceWithoutDiscount() {
    Customer BigCo = new Customer("BigCo");

    Play Hamlet = new Tragedy("Hamlet");
    Play AsYouLikeIt = new Comedy("As You Like It");
    Play Othello = new Tragedy("Othello");

    Invoice invoice = new Invoice(BigCo, List.of(
            new Performance(Hamlet, 55),
            new Performance(AsYouLikeIt, 35),
            new Performance(Othello, 40)));

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.printTXT(invoice.getData());

    verify(result);
  }

  @Test
  void testTXTInvoiceWithDiscount() {
    Customer BigCo = new Customer("BigCo");

    BigCo.credits = 130;

    Play Hamlet = new Tragedy("Hamlet");
    Play AsYouLikeIt = new Comedy("As You Like It");
    Play Othello = new Tragedy("Othello");

    Invoice invoice = new Invoice(BigCo, List.of(
      new Performance(Hamlet, 55),
      new Performance(AsYouLikeIt, 35),
      new Performance(Othello, 40)));

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.printTXT(invoice.getData());

    verify(result);
  }

  @Test
  void testHTMLInvoiceWithoutDiscount() {
    Customer BigCo = new Customer("BigCo");

    Play Hamlet = new Tragedy("Hamlet");
    Play AsYouLikeIt = new Comedy("As You Like It");
    Play Othello = new Tragedy("Othello");

    Invoice invoice = new Invoice(BigCo, List.of(
      new Performance(Hamlet, 55),
      new Performance(AsYouLikeIt, 35),
      new Performance(Othello, 40)));

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.printHTML(invoice.getData());

    verifyHtml(result);
  }

  @Test
  void testHTMLInvoiceWithDiscount() {
    Customer BigCo = new Customer("BigCo");

    BigCo.credits = 130;

    Play Hamlet = new Tragedy("Hamlet");
    Play AsYouLikeIt = new Comedy("As You Like It");
    Play Othello = new Tragedy("Othello");

    Invoice invoice = new Invoice(BigCo, List.of(
      new Performance(Hamlet, 55),
      new Performance(AsYouLikeIt, 35),
      new Performance(Othello, 40)));

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.printHTML(invoice.getData());

    verifyHtml(result);
  }
}
