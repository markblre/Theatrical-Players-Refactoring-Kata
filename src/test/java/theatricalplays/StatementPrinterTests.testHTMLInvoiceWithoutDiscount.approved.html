<h1>Statement for BigCo</h1>
<table border='1'>
  <tr><th>play</th><th>seats</th><th>price</th></tr>
  <tr><td>Hamlet</td><td>55</td><td>$650.00</td></tr>
  <tr><td>As You Like It</td><td>35</td><td>$580.00</td></tr>
  <tr><td>Othello</td><td>40</td><td>$500.00</td></tr>
</table>
<p>Subtotal: $1,730.00</p>
<p>Previous credit balance: 0</p>
<p>You earned <em>47</em> credits</p>
<p>New credit balance: 47</p>
<p>Total: <em>$1,730.00</em></p>
